### PR TITLE
Basic documentation for readthedocs

### PR DESCRIPTION
**Pull request by jchodera**
This PR creates basic sphinx documentation for readthedocs.

Note that this required removing an earlier aborted attempt (which contained nothing useful).

Since so many files have changed, please just review the [rendered docs](http://openmmtools.readthedocs.io/) and the [updated README](https://github.com/choderalab/openmmtools/blob/6ea59619f2b38bc9fe6282b6f74972cdfebc47fd/README.md).

### CODE REVIEW

**Comment by jchodera** [comment]
And here's the rendered product!

http://openmmtools.readthedocs.io/

File: docs/index.rst [hunk 1]
```diff
@@ -1,26 +1,61 @@
...
+    This is module is undergoing heavy development. None of the API calls are final.
+    This software is provided without any guarantees of correctness, you will likely encounter bugs.
+
+    If you are interested in this code, please wait for the official release to use it.
+    In the mean time, to stay informed of development progress you are encouraged to:
+
+    * Follow `this feed`_ for updates on releases.
+    * Check out the `github repository`_ .
+
+.. _this feed: https://github.com/choderalab/constph-openmm/releases.atom
```
Comment (kyleabeauchamp) [inline docs/index.rst]: Points to wrong repo here?

Comment (jchodera) [inline docs/index.rst]: Whoops, thanks!

**Comment by jchodera** [comment]
What's the best way to include a list of authors/contributors? Is there some way to pull this in from github?

**Comment by jchodera** [comment]
This should be ready to merge now.

The docstrings can probably use a lot of cleanup in a separate PR.
We'll especially have to figure out how signatures for subclasses that just use `*args, **kwargs` to pass arguments up to base classes could provide more informative signatures.

**Comment by jchodera** [comment]
Here's the updated [README](https://github.com/choderalab/openmmtools/blob/6ea59619f2b38bc9fe6282b6f74972cdfebc47fd/README.md).

File: docs/installation.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+On ``linux``, you can install the Python 3 version into ``$HOME/miniconda3`` with (on ``bash`` systems):
+
+.. code-block:: bash
+
+   $ wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+   $ bash ./Miniconda3-latest-Linux-x86_64.sh -b -p $HOME/miniconda3
+   $ export PATH="$HOME/miniconda3/bin:$PATH"
+
+On ``osx``, you want to use the ```osx`` binary
```
Comment (bas-rustenburg) [inline docs/installation.rst]: One ` too many

**Comment by bas-rustenburg** [comment]
image

Might need to edit the highlighted section (not sure where in the files this occurs).

**Comment by bas-rustenburg** [comment]
http://openmmtools.readthedocs.io/en/latest/api/generated/openmmtools.testsystems.HostGuestImplicitHCT.html#openmmtools.testsystems.HostGuestImplicitHCT

Once you go into an API page, there doesn't seem to be a link to click back into another part of the page. The user can press the back button in the browser but could be convenient to have the table of content link there like on the other, regular pages. 

Minor point, and not sure if this is easy to achieve.

**Comment by jchodera** [comment]
> Might need to edit the highlighted section (not sure where in the files this occurs).

These are docstring issues. The docstrings need some major cleanup, but that's beyond the scope of this PR.

> Minor points only. Looks like the structure is there. We can refine the content as we go. Should be helpful.

That's exactly my hope.

> Once you go into an API page, there doesn't seem to be a link to click back into another part of the page. The user can press the back button in the browser but could be convenient to have the table of content link there like on the other, regular pages.

I have no idea how to do this with sphinx, but if you can figure it out, that would make a great other PR.

**Comment by jchodera** [comment]
Merging this since I'm having trouble fixing the typos by pushing to this branch.

**Pull request merged**